Core/Utilities/SiteSEMapping.py

```diff
@@ -8,10 +8,86 @@
 
 __RCSID__ = "$Id$"
 
-from DIRAC import S_OK
+import errno
+
+from DIRAC import gLogger, S_OK, S_ERROR
 from DIRAC.DataManagementSystem.Utilities.DMSHelpers import DMSHelpers, siteGridName
 
 
+def getSEParameters(seName, plugins=None):
+  """ get all the SE parameters in a list
+
+      :param str seName: name of the Storage Element
+      :param list plugins: if provided, restrict to a certain list of plugins
+
+      :return: S_OK() or S_ERROR
+  """
+  # This import is here to avoid circular imports
+  from DIRAC.Resources.Storage.StorageElement import StorageElement
+  se = StorageElement(seName, hideExceptions=True)
+
+  seParameters = S_ERROR(errno.ENODATA, 'No SE parameters obtained')
+  pluginsList = se.getPlugins()
+  if not pluginsList['OK']:
+    gLogger.warn(pluginsList['Message'], "SE: %s" % seName)
+    return pluginsList
+  if plugins:
+    pluginsSet = set(pluginsList['Value']).intersection(set(plugins))
+  else:
+    pluginsSet = set(pluginsList['Value'])
+
+  seParametersList = []
+  for plugin in pluginsSet:
+    seParameters = se.getStorageParameters(plugin)
+    if seParameters['OK']:
+      seParametersList.append(seParameters['Value'])
+
+  return S_OK(seParametersList)
+
+
+def getSEHost(seName, plugins=None):
+  """ Get StorageElement host names (can be more than one depending on the protocol)
+
+      :param str seName: name of the storage element
+      :param list plugins: if provided, restrict to a certain list of plugins
+
+      :return: S_OK() with list of hosts or S_ERROR
+  """
+
+  seParameters = getSEParameters(seName, plugins)
+  if not seParameters['OK']:
+    gLogger.warn("Could not get SE parameters", "SE: %s" % seName)
+    return seParameters
+
+  return S_OK([parameters['Host'] for parameters in seParameters['Value']])
+
+
+def getStorageElementsHosts(seNames=None, plugins=None):
+  """ Get StorageElement host names
+
+      :param list seNames: possible list of storage element names (if not provided, will use all)
+      :param list plugins: if provided, restrict to a certain list of plugins
+
+      :return: S_OK() with list of hosts or S_ERROR
+  """
+
+  seHosts = []
+
+  if seNames is None:
+    seNames = DMSHelpers().getStorageElements()
+
+  for seName in seNames:
+
+    seHost = getSEHost(seName, plugins)
+    if not seHost['OK']:
+      gLogger.warn("Could not get SE Host", "SE: %s" % seName)
+      continue
+    if seHost['Value']:
+      seHosts.append(seHost['Value'])
```

Comment:
append or extend ?

Reply:
Good catch. Changed here and in another place. And I changed the name to `getSEHosts`